frontend/src/main/scala/bloop/bsp/BloopBspServices.scala

```diff
@@ -266,7 +276,9 @@ final class BloopBspServices(
     callSiteState.logger.info("BSP initialization handshake complete.")
   }
 
-  def ifInitialized[T](compute: BspComputation[T]): BspEndpointResponse[T] = {
+  def ifInitialized[T](
+      bspLogger: BspServerLogger
```

Comment:
maybe it could accept an `originId: Option[String]` ? that way each invocation would only have to consider if the originId is in scope.